src/billforward.js

```diff
@@ -2947,6 +2972,73 @@
         return TheClass;
     })();
 
+    bfjs.GoCardlessTransaction = (function() {
+        const _parent = bfjs.GatewayTransaction;
+
+        const TheClass = function() {
+            _parent.apply(this, arguments);
+        };
+
+        const p = TheClass.prototype = new _parent();
+        p.constructor = TheClass;
+
+        TheClass.construct = (function() {
+            // factory pattern for invoking own constructor with arguments
+            // basically: return new this(arguments)
+
+            function lambda(args) {
+                return TheClass.apply(this, args);
+            }
+            lambda.prototype = TheClass.prototype;
+
+            return function() {
+                return new lambda(arguments);
+            }
+        })();
+
+        p['do'] = function() {
+            // do nothing
+        };
+
+        p.doSubmitDanceWhenReady = function () {
```

Comment:
Is the term `dance` used elsewhere in this repo? If not, maybe it should be more official-looking?

Reply:
Yeah, this function name is also used for the other payment gateways. I'm not convinced it's a good name, but I wanted to be consistent.